src/control/server/storage/config.go

```diff
@@ -259,6 +267,177 @@ func (sc *ScmConfig) Validate(class Class) error {
 	return nil
 }
 
+type stringSet map[string]struct{}
+
+func (ss stringSet) Add(s string) error {
+	if _, exists := ss[s]; exists {
+		return errors.Errorf("duplicate device %q", s)
+	}
+	ss[s] = struct{}{}
+	return nil
+}
+
+func (ss stringSet) Strings() []string {
+	var out []string
+	for s := range ss {
+		out = append(out, s)
+	}
+	sort.Strings(out)
+	return out
+}
+
+// BdevDeviceList represents a set of block device addresses.
+type BdevDeviceList struct {
```

Comment:
From what I understand, I see one drawbacks with the proposed way of dealing with non PCI address. If we are mixing PCI bdev and non PCI bdev, then the non PCI bdev will not be accessible. Maybe, we should not allow to mix the two kinds of Device when created with the fromStrings method.